### PR TITLE
zones: check all custom zones at point

### PR DESCRIPTION
**Pull request by andrei8l**
#### Summary
None

#### Purpose of change
The zone manager only checks the first zone of a type at a point. For `LOOT_CUSTOM` and `LOOT_ITEM_GROUP`, we want to check all zones since they can have different parameters.

#### Describe the solution
Check all the zones at point

#### Describe alternatives you've considered
N/A

#### Testing
Unit test

<details>
<summary>Manual testing</summary>

Assuming #58150 has not been merged yet:
Go to Jay Ruckers' shop in `isolated_road` and confirm that a rifle was distributed on the table to the north rather than in the shelves to the east.
</details>

#### Additional context
Discovered while fumbling about with #58150

### CODE REVIEW

**Comment by bombasticSlacks** [comment]
Thanks for looking at this!

File: src/clzones.h [hunk 1]
```diff
@@ -440,8 +440,12 @@ class zone_manager
                   const tripoint &start, const tripoint &end,
                   const shared_ptr_fast<zone_options> &options = nullptr, const bool personal = false,
                   bool silent = false, map *pmap = nullptr );
+        // get first matching zone
```
Comment (BevapDin) [inline src/clzones.h:443]: It's returning `ret.back();`, so it's actually the last matching zone, is it not?

Comment (andrei8l) [reply src/clzones.h:443]: Good catch. Thanks!

**Comment by andrei8l** [comment]
Thanks for merging

**Pull request merged**